terraform/batch-processing/src/batch_processing.py

```diff
@@ -1,20 +1,84 @@
 from time import time
 from boto3 import resource
+import http.client
+import json
+import uuid
 
 
 def lambda_handler(_event, _context):
-    source_bucket_name = _event.get("Records")[0].get("s3").get("bucket").get("name")
-    dest_bucket_name = source_bucket_name.replace("source", "destination")
-    output_bucket = resource('s3').Bucket(dest_bucket_name)
+    # Check if "Records" key exists and it's a non-empty list
+    if "Records" in _event and isinstance(_event["Records"], list) and len(_event["Records"]) > 0:
+        s3_record = _event["Records"][0]["s3"]
 
-    # Write some placeholder bytestring data to a file in the bucket,
-    # so we can test that the lambda writes to the correct output bucket.
-    filename = f"output_report_{time()}.txt"
-    data = (b'Test file to see if the lambda writes to the correct s3 bucket. '
-            b'If our AWS bill skyrockets, this file has been written to the wrong bucket!')
+        # Check if "bucket" key exists and it's a dictionary
+        if "bucket" in s3_record and isinstance(s3_record["bucket"], dict):
+            source_bucket_name = s3_record["bucket"].get("name")
 
-    output_bucket.put_object(Body=data, Key=filename)
+            # Check if "object" key exists and it's a dictionary
+            if "object" in s3_record and isinstance(s3_record["object"], dict):
+                object_key = s3_record["object"].get("key")
 
+                source_bucket_name = _event.get("Records")[0].get("s3").get("bucket").get("name")
+                dest_bucket_name = source_bucket_name.replace("source", "destination")
+                output_bucket = resource('s3').Bucket(dest_bucket_name)
+                api_gateway_url = "https://hphlz3bvpb.execute-api.eu-west-2.amazonaws.com/alli5"
+
+            # getting JSON object
+                s3_record = _event["Records"][0]["s3"]
+                object_key = s3_record["object"]["key"]
+
+            # download JSON object from S3
+                s3_object_url = f"s3://{source_bucket_name}/{object_key}"
+                conn = http.client.HTTPSConnection(api_gateway_url)
+                conn.request("GET", s3_object_url)
+                response = conn.getresponse()
+
+            # send JSON object to API-gateway if successfully downloaded
+                if response.status == 200:
+                    json_data = json.loads(response.read().decode('utf-8'))
```

Comment:
We're expecting a csv file inside that s3 bucket. You can just assume (hardcode) an array of immunisation objects. 
Imagine a function where you send the content of your s3 bucket and you get a list of json objects back. For now, write it such a way that the list is hardcoded. Almas' ticket will fill that function. 
Now you have a list of immunisation objects. So you need a for loop to perform one single request per object. logging each request whilst making a record of failures so after the loop has finished you can put it into dest bucket.